Autopilot/PathManager/Src/landingManager.cpp

```diff
@@ -0,0 +1,202 @@
+#include "waypointManager.hpp"
+#include "landingManager.hpp"
+#include <math.h>
+
+const static int SET_THROTTLE_OFF = -1000;
+const static double DISTANCE_OF_LANDING = 10; //in meters
+const static int METERS_PER_DEG_LAT = 111320;
+const static int ANGLE_OF_LANDING = 5; //in degrees
+
+const static double PI = 3.14159265358979323846;
+
+double LandingManager::changingAltitude(_WaypointManager_Data_In * input, _PathData * aimingPoint, _PathData * intersectionPoint, _PathData * stoppingPoint)
+{
+    //arrays to hold the important points
+    float aPoint[3]; //aiming point
+    float iPoint[3]; //intersection point
+    float sPoint[3]; //stopping point
+    float cPoint[3]; //current point
+    float projectedPoint[3];
+
+    //loading important points
+    aPoint[0] = aimingPoint->longitude;
+    aPoint[1] = aimingPoint->latitude;
+    aPoint[2] = aimingPoint->altitude; 
+
+    iPoint[0] = intersectionPoint->longitude;
+    iPoint[1] = intersectionPoint->latitude;
+    iPoint[2] = intersectionPoint->altitude;
+
+    sPoint[0] = stoppingPoint->longitude;
+    sPoint[1] = stoppingPoint->latitude;
+    sPoint[2] = stoppingPoint->altitude;
+
+    cPoint[0] = input->longitude;
+    cPoint[1] = input->latitude;
+    cPoint[2] = input->altitude;
+
+    //calculating the vectors
+    float vectorAI[3];
+    float vectorAS[3];
+    float vectorAC[3];
+    float normal[3];
```

Comment:
I think the changingAltitude function looks fine, as ur doing the vector operations exactly as u said in the description. I'm just wondering why the 'normal' array is highlighted? I tried googling it but it doesn't seem to be a keyword?

Reply:
Hmm I'm not sure, I gotta look into that. In VS code it doesn't show up as a keyword, idk why its showing up as a keyword in git.

Reply:
Yeah I tried it in VScode too, it didn't highlight weirdly for me either